File: routeMap/app/src/main/java/ru/android/zheka/coreUI/ErrorControl.java
package ru.android.zheka.coreUI;

import io.reactivex.functions.Consumer;
import ru.android.zheka.gmapexample1.R;

public class ErrorControl {
    private IActivity view;

    public ErrorControl(IActivity view) {
        this.view = view;
    }

    public void showError(Throwable throwable,  Consumer<Boolean> consumer) {
        new ErrorDialog (DialogConfig.builder ().contentId (R.layout.alert_dialog)
                .context (view.getContext ())
                .poistiveBtnId (R.string.ok_save_point)
                .build (), view).showError (throwable, consumer);
    }
}

File: routeMap/app/src/main/java/ru/android/zheka/coreUI/ErrorControl.kt
package ru.android.zheka.coreUI

import io.reactivex.functions.Consumer
import ru.android.zheka.gmapexample1.R

class ErrorControl(private val view: IActivity) {
    fun showError(throwable: Throwable, consumer: Consumer<Boolean>) {
        ErrorDialog(DialogConfig.Companion.builder().contentId(R.layout.alert_dialog)
                .context(view.activity)
                .poistiveBtnId(R.string.ok_save_point)
                .build(), view).showError(throwable, consumer)
    }

}